Mathlib/Topology/CompactOpen.lean

```diff
@@ -255,6 +255,25 @@ instance [T3Space Y] : T3Space C(X, Y) := inferInstance
 
 end Ev
 
+section DiscreteTopology
+variable [DiscreteTopology X]
+
+/-- The continuous functions from `X` to `Y` are the same as the plain functions when `X` is
+discrete. -/
+@[simps!]
+def homeoFnOfDiscrete : C(X, Y) ≃ₜ (X → Y) where
+  __ := equivFnOfDiscrete
+  continuous_invFun :=
+    continuous_compactOpen.2 fun K hK U hU ↦ isOpen_set_pi hK.finite_of_discrete fun _ _ ↦ hU
+
+@[simp] lemma coe_homeoFnOfDiscrete : ⇑homeoFnOfDiscrete = (DFunLike.coe : C(X, Y) → X → Y) := rfl
+
+@[simp] lemma homeoFnOfDiscrete_symm_apply (f : X → Y) : homeoFnOfDiscrete.symm f = f := rfl
```

Comment:
Are these lemmas the same as those generated by `simps!` above? What if you pass `-fullyApplied`?

Reply:
Now you have both `simps` and hand-written lemmas. Is it intentional? Anyway,
bors d+
One more optional suggestion: add a `@[simp]` lemma for the `toEquiv` projection.